DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/EditDelegateController.cs

```diff
@@ -0,0 +1,112 @@
+﻿namespace DigitalLearningSolutions.Web.Controllers.TrackingSystem.Delegates
+{
+    using System.Linq;
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Attributes;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.Models.Enums;
+    using DigitalLearningSolutions.Web.ViewModels.Common;
+    using DigitalLearningSolutions.Web.ViewModels.TrackingSystem.Delegates.EditDelegate;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.FeatureManagement.Mvc;
+
+    [FeatureGate(FeatureFlags.RefactoredTrackingSystem)]
+    [Authorize(Policy = CustomPolicies.UserCentreAdmin)]
+    [Route("TrackingSystem/Delegates/{delegateId:int}/Edit")]
+    [SetDlsSubApplication(nameof(DlsSubApplication.TrackingSystem))]
+    public class EditDelegateController : Controller
+    {
+        private readonly CentreCustomPromptHelper centreCustomPromptHelper;
+        private readonly IJobGroupsDataService jobGroupsDataService;
+        private readonly IUserService userService;
+
+        public EditDelegateController(
+            IUserService userService,
+            IJobGroupsDataService jobGroupsDataService,
+            CentreCustomPromptHelper customPromptHelper
+        )
+        {
+            this.userService = userService;
+            this.jobGroupsDataService = jobGroupsDataService;
+            centreCustomPromptHelper = customPromptHelper;
+        }
+
+        [HttpGet]
+        public IActionResult Index(int delegateId)
+        {
+            var centreId = User.GetCentreId();
+            var delegateUser = userService.GetUsersById(null, delegateId).delegateUser;
+
+            if (delegateUser == null || delegateUser.CentreId != centreId)
+            {
+                return NotFound();
+            }
+
+            var jobGroups = jobGroupsDataService.GetJobGroupsAlphabetical().ToList();
+
+            var customPrompts =
+                centreCustomPromptHelper.GetEditCustomFieldViewModelsForCentre(delegateUser, centreId);
+            var model = new EditDelegateViewModel(delegateUser, jobGroups, customPrompts);
+
+            return View(model);
+        }
+
+        [HttpPost]
+        public IActionResult Index(EditDelegateFormData formData, int delegateId)
+        {
+            var centreId = User.GetCentreId();
+            if (!formData.JobGroupId.HasValue)
+            {
+                ModelState.AddModelError(nameof(EditDetailsFormData.JobGroupId), "Select a job group");

```

Comment:
You should be able to do this in the View Model (I'm not 100% on exactly how that should work with the Form Data structure but at the very least a custom Validate method on EditDelegateFormData will do it I think)